src/laser/MultiLaser.cpp

```diff
@@ -534,50 +589,216 @@ MultiLaser::ShiftLaserSlices ()
+
+        // put chi from the plasma density function on the laser grid as if it were deposited there,
+        // this works even outside the field grid
+        // note that the effect of temperature / non-zero u is ignored here
+        for (auto& plasma : multi_plasma.m_all_plasmas) {
+
+            const PhysConst pc = get_phys_const();
+            const amrex::Real c_t = pc.c * Hipace::m_physical_time;
+            amrex::Real chi_factor = plasma.GetCharge() * plasma.GetCharge() * pc.mu0 / plasma.GetMass();
+            if (plasma.m_can_ionize) {
+                chi_factor *= plasma.m_init_ion_lev * plasma.m_init_ion_lev;
+            }
+
+            auto density_func = plasma.m_density_func;
+
+            const amrex::Real poff_laser_x = GetPosOffset(0, m_laser_geom_3D, m_laser_geom_3D.Domain());
+            const amrex::Real poff_laser_y = GetPosOffset(1, m_laser_geom_3D, m_laser_geom_3D.Domain());
+
+            const amrex::Real dx_laser = m_laser_geom_3D.CellSize(0);
+            const amrex::Real dy_laser = m_laser_geom_3D.CellSize(1);
+
+            amrex::ParallelFor(mfi.growntilebox(),
+                [=] AMREX_GPU_DEVICE(int i, int j, int) noexcept {
+                    const amrex::Real x = i * dx_laser + poff_laser_x;
+                    const amrex::Real y = j * dy_laser + poff_laser_y;
+
+                    laser_arr_chi(i, j) += density_func(x, y, c_t) * chi_factor;
+                });
+        }
+    }
+}
+
+void
+MultiLaser::InterpolateChi (const Fields& fields, amrex::Geometry const& geom_field_lev0)
+{
+    HIPACE_PROFILE("MultiLaser::InterpolateChi()");
+
+    for ( amrex::MFIter mfi(m_slices, DfltMfi); mfi.isValid(); ++mfi ){
+        Array3<amrex::Real> laser_arr = m_slices.array(mfi);
+        Array2<const amrex::Real> field_arr_chi =
+            fields.getSlices(0).array(mfi, Comps[WhichSlice::This]["chi"]);
+
+        const amrex::Real poff_laser_x = GetPosOffset(0, m_laser_geom_3D, m_laser_geom_3D.Domain());
+        const amrex::Real poff_laser_y = GetPosOffset(1, m_laser_geom_3D, m_laser_geom_3D.Domain());
+        const amrex::Real poff_field_x = GetPosOffset(0, geom_field_lev0, geom_field_lev0.Domain());
+        const amrex::Real poff_field_y = GetPosOffset(1, geom_field_lev0, geom_field_lev0.Domain());
+
+        const amrex::Real dx_laser = m_laser_geom_3D.CellSize(0);
+        const amrex::Real dy_laser = m_laser_geom_3D.CellSize(1);
+        const amrex::Real dx_laser_inv = m_laser_geom_3D.InvCellSize(0);
+        const amrex::Real dy_laser_inv = m_laser_geom_3D.InvCellSize(1);
+        const amrex::Real dx_field = geom_field_lev0.CellSize(0);
+        const amrex::Real dy_field = geom_field_lev0.CellSize(1);
+        const amrex::Real dx_field_inv = geom_field_lev0.InvCellSize(0);
+        const amrex::Real dy_field_inv = geom_field_lev0.InvCellSize(1);
+
+        amrex::Box field_box = fields.getSlices(0)[mfi].box();
+        field_box.grow(-2*Fields::m_slices_nguards);
```

Comment:
Could you add a comment to explain this grow? I think the explanation below is OK, but feel free to fix if not.
```suggestion
        # Even in the valid domain, chi near the boundaries is incorrect due to finite deposition order.
        field_box.grow(-2*Fields::m_slices_nguards);
```